source/iterator/iterator.go

```diff
@@ -0,0 +1,188 @@
+// Copyright © 2022 Meroxa, Inc. & Yalantis
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iterator
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/jmoiron/sqlx"
+)
+
+// metadata related.
+const (
+	metadataTable = "clickhouse.table"
+
+	querySelectRowsFmt = "SELECT %s FROM %s%s ORDER BY %s ASC LIMIT %d;"
+	whereClauseFmt     = " WHERE %s > ?"
+)
+
+// Iterator represents an implementation of an iterator for ClickHouse.
+type Iterator struct {
+	db   *sqlx.DB
+	rows *sqlx.Rows
+
+	// represents a position
+	lastProcessedVal any
+	// represents a table name
+	table string
+	// represents a name of column what iterator use for setting key in record
+	keyColumns []string
+	// represents a name of column what iterator use for sorting data
+	orderingColumn string
+	// represents a list of table's columns for record payload.
+	// if empty - will get all columns
+	columns []string
+	// represents a size of batch
+	batchSize int
+}
+
+// Params represents an incoming iterator params for the New function.
+type Params struct {
+	DB               *sqlx.DB
+	LastProcessedVal any
+	URL              string
```

Comment:
It seems that you don't use this parameter anywhere.

Reply:
You're right.
I've removed it.